### ISSUE TITLE
Can't import Tandoor export zip file

### ISSUE THREAD

**Issue by alexbonazzi** [TandoorRecipes/recipes#2433]
### Tandoor Version

1.4.7_11.0.34 on TrueNAS scale

### Setup

Kubernetes

### Reverse Proxy

No reverse proxy

### Bug description

I reinstalled tandoor on TrueNAS-SCALE-22.12.2.
Whn I try to import the default export from my PC I got this error "There was an error creating a resource!" and no details

**Comment by vabene1111** [comment]
have you exported in the same version you are trying to import into?

**Comment by alexbonazzi** [comment]
No Ihad to export alle recipes and reinstall cause Truechart don't upgrade anymore apps without reinstallation

**Comment by alexbonazzi** [comment]
I didn't wrote the version I think the  precedent

**Comment by vabene1111** [comment]
generally import/export is not meant as a way to backup your data. You will loose quite a bit of stuff if you do it this way. If your installation does not give you another option i recommend dumping the database and loading it after upgrading. The migrations will be applied automatically (at least if you use docker)

**Comment by alexbonazzi** [comment]
I sent you data. The database is MySQL ? How can I connect to database ?  
Loading data is really great help cause I can see all recipes and complete them.

**Comment by alexbonazzi** [comment]
I checked some things.
MY version was the same cause TrueNAs said that there is no need for update.
I exctraced one recipe and tried to import. I got the same error but also this detail "no zip file"

**Comment by vabene1111** [comment]
need to look at that, generally i cannot support mysql or truenas installations since i dont know much about that, i can see if there is an actual bug in the importer, if so I can fix it.

**Comment by alexbonazzi** [comment]
Ok it seems that when I specify the backup zip it is not uploaded by the app.

**Comment by alexbonazzi** [comment]
I can only select a local file and I can't write a filename

**Comment by vabene1111** [comment]
so i just tried importing your file, everything works fine. I am getting 74 recipes and all that i looked at seem to be imported correctly. This must be an issue with your installation i guess.

i tested in the just released 1.4.9 version

**Comment by alexbonazzi** [comment]
The last version released by Truecharts is 1.4.
The backup zip is located my PC, May be that is not loaded by the app. Can you suggest how can I check it ? 
Thanks

**Comment by vabene1111** [comment]
The system page can tell you the latest version. If it's 1.4. 0 it's quite old, might be that there was a bug back then

**Comment by alexbonazzi** [comment]
Version: is 1.4.7
Ref: [82d1a75d803631493d78e80d3db161787d78b921]

**Comment by vabene1111** [comment]
i am sorry but I dont know what else to do. Your export works perfectly fine for me and between 1.4.7. and the current version not much has changed in that regard that I would remember.

**Comment by vabene1111** [comment]
The application loads it directly into memory. You can set debug to 1 in the environment file, maybe that will show logs in the console

**Comment by alexbonazzi** [comment]
ok "debug" lower case ?
thanks

**Comment by vabene1111** [comment]
No upper case, it's in the environment template

**Comment by alexbonazzi** [comment]
This is log line
023-04-28 16:25:52.266398+00:002023-04-28 16:25:52.266 GMT [28723] LOG:  could not receive data from client: Connection reset by peer

**Comment by alexbonazzi** [comment]
I tried with Chrome and Opera

**Comment by vabene1111** [comment]
No idea,sorry.have never used your setup method, there could be a million things wrong there.

**Comment by alexbonazzi** [comment]
For the next time how can I backup database ?
where is it ?

**Comment by vabene1111** [comment]
usually its a postgres container, backup there is pretty easy but i dont know what you did with true nas so no idea how to do that. If its individual containers it would probably be easiest to just update the application and leave the database container untouched

**Comment by alexbonazzi** [comment]
I found also this in log

2023-04-30 16:29:25.079491+00:002023/04/30 16:29:25 [error] 29#29: *596 client intended to send too large body: 19700684 bytes, client: 192.168.2.7, server: _, request: "POST /api/import/ HTTP/1.1", host: "192.168.2.5:10029", referrer: "http://192.168.2.5:10029/data/import/url"


 where can I increase maximum upload limit I can't find it in environnement variabiles ?

**Comment by smilerz** [comment]
It’s in the nginx config.  You might be better off splitting the import in two though.

**Comment by alexbonazzi** [comment]
I splitted in five pieces and I upload all recipes, thanks.
My TrueNAS app installation don't have anything about nginx reverse proxy, where can i find it ?

**Comment by vabene1111** [comment]
something needs to serve the application on your true nas and there needs to be an option to set the upload size there

thinking about solutions to handle this error in the application better but i think it wont be possible since the request wont be seen by the application 🤔

**Comment by alexbonazzi** [comment]
If  can suggest you it will be nice if it will be possible to backup and restore recipes in a directory, as separate files.
It will also nice to backup and restore all data.

**Comment by alexbonazzi** [comment]
If can suggest you it will be nice if it will be possible to backup and restore recipes in a directory, as separate files.
It will also be nice to backup and restore all data.

**Comment by vabene1111** [comment]
yes, this has been on the wishlist for quite some time but given that tandoor still regularely develops and changes its quite a bit work to maintain this troguh all versions so for now its database backups only

**Comment by maku1958** [comment]
I encountered the same problem. I have 36 Recipies which I exported end of April. I would like to import them but tandoor claims the files are not a zip file. Can I send you my zip folder ? But I didn't dare to import all zips at once. I started with one after the other each one seems to be not a zip file.
There are some recipes from Chefkoch.de can I upload the zip here or better not?
I better send the folder to the email address mentioned above.
Thanks in advance.
Manfred
